src/overlays/actors/ovl_Dm_Ravine/z_dm_ravine.c

```diff
@@ -6,29 +12,68 @@
 
 void DmRavine_Init(Actor* thisx, GlobalContext* globalCtx);
 void DmRavine_Destroy(Actor* thisx, GlobalContext* globalCtx);
+void DmRavine_Action(Actor* thisx, GlobalContext* globalCtx);
 void DmRavine_Update(Actor* thisx, GlobalContext* globalCtx);
 void DmRavine_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
-const ActorInit Dm_Ravine_InitVars = {
-    ACTOR_DM_RAVINE,
-    ACTORCAT_ITEMACTION,
-    FLAGS,
-    OBJECT_KEIKOKU_DEMO,
-    sizeof(DmRavine),
-    (ActorFunc)DmRavine_Init,
-    (ActorFunc)DmRavine_Destroy,
-    (ActorFunc)DmRavine_Update,
-    (ActorFunc)DmRavine_Draw
-};
-*/
+const ActorInit Dm_Ravine_InitVars = { ACTOR_DM_RAVINE,
+                                       ACTORCAT_ITEMACTION,
+                                       FLAGS,
+                                       OBJECT_KEIKOKU_DEMO,
+                                       sizeof(DmRavine),
+                                       (ActorFunc)DmRavine_Init,
+                                       (ActorFunc)DmRavine_Destroy,
+                                       (ActorFunc)DmRavine_Update,
+                                       (ActorFunc)DmRavine_Draw };
+
+void DmRavine_Init(Actor* thisx, GlobalContext* globalCtx) {
+    u8 flag;
+
+    flag = gSaveContext.perm.weekEventReg[0];
+    if (((flag & 0x10) | cREG(0)) != 0) {
+        Actor_MarkForDeath(thisx);
+        return;
+    }
+
+    THIS->loaded = false;
```

Comment:
instead of THIS->.... make a temp before `u8 flag` that is `DmRavine* this = THIS;`, and then just use `this->...`